dcor/independence.py

```diff
@@ -347,3 +356,105 @@ def distance_correlation_t_test(
     p_value = 1 - scipy.stats.t.cdf(t_test, df=df)
 
     return HypothesisTest(pvalue=p_value, statistic=t_test)
+
+
+##-----------------------------------------------------------------------------------------------------------------------
+''' 
+A Statistically and Numerically Efficient Independence Test Based on Random Projections and Distance Covariance
+
+url: https://www.frontiersin.org/articles/10.3389/fams.2021.779841/full
+'''
+
+mp.dps = 25; mp.pretty = True
+def gamma_cdf(x, shape,  scale):
+    return gammainc(shape, a = 0, b = float(x/scale))/np.exp(gammaln(shape))
```

Comment:
You can use [`scipy.special.gammainc`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.gammainc.html), so that `mpmath` is not required.

Reply:
I have changed the code as you suggested, but based on my experimental outcomes mpmath helps to calculate accurate integration compare to scipy.stats one